tpgtools/sample.go

```diff
@@ -249,6 +256,12 @@ func (s *Sample) EnumerateWithUpdateSamples() []Sample {
 			var newDeps []Dependency
 			newDeps = append(newDeps, newSample.DependencyList...)
 			newDeps[0] = newSample.generateSampleDependencyWithName(*newSample.PrimaryResource, "primary")
+
+			// add the old primary resource to the list
+			oldPrimaryResource := newSample.generateSampleDependency(*s.PrimaryResource)
```

Comment:
won't this resource also need to have the name 'primary` so the reference that replaces it is correct?

```suggestion
			oldPrimaryResource := newSample.generateSampleDependencyWithName(*s.PrimaryResource, "primary")
```

Reply:
it shouldn't matter either way, `primary` here is just used as the HCL block name. Like `resource "google_compute_firewall_policy" "primary" {`. Maybe there's a better way to indicate what it does.

Reference substitution looks for the filename: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/tpgtools/sample.go#L182

Reply:
talked offline. @ScottSuarez's right, this won't work as a reference